src/app/model/ChessEngine.ts

```diff
@@ -99,8 +111,107 @@ export class ChessEngine implements ChessModel {
     }
   }
 
+  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
+    const chosenPiece = this.board.onPositionPiece(squareFrom);
+    const whiteCastlingFrom: Square = { column: 'E', row: 1 };
+    const whiteLongCastlingTo: Square = { column: 'C', row: 1 };
+    const whiteShortCastlingTo: Square = { column: 'G', row: 1 };
+    const blackCastlingFrom: Square = { column: 'E', row: 8 };
+    const blackLongCastlingTo: Square = { column: 'C', row: 8 };
+    const blackShortCastlingTo: Square = { column: 'G', row: 8 };
+
+    return (
```

Comment:
Jak dla mnie łatwiej byłoby się połapać np. mniej więcej tak, w tych ifach:
```.ts
  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
    const chosenPiece = this.board.onPositionPiece(squareFrom);
    if (!(chosenPiece instanceof King)) {
      return false;
    }

    const whiteCastlingFrom: Square = { column: 'E', row: 1 };
    const whiteLongCastlingTo: Square = { column: 'C', row: 1 };
    const whiteShortCastlingTo: Square = { column: 'G', row: 1 };
    const blackCastlingFrom: Square = { column: 'E', row: 8 };
    const blackLongCastlingTo: Square = { column: 'C', row: 8 };
    const blackShortCastlingTo: Square = { column: 'G', row: 8 };

    const castlingAvailableForWhiteKing = chosenPiece.side === Side.WHITE 
      && this.isSameSquare(squareFrom, whiteCastlingFrom) 
      && (this.isSameSquare(squareTo, whiteLongCastlingTo) || this.isSameSquare(squareTo, whiteShortCastlingTo));
    const castlingAvailableForBlackKing = chosenPiece.side === Side.BLACK 
      && this.isSameSquare(squareFrom, blackCastlingFrom) 
      && (this.isSameSquare(squareTo, blackLongCastlingTo) || this.isSameSquare(squareTo, blackShortCastlingTo));
    return castlingAvailableForWhiteKing || castlingAvailableForBlackKing;
  }
```

Reply:
Albo idąc dalej:
```.ts
  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
    const chosenPiece = this.board.onPositionPiece(squareFrom);
    if (!(chosenPiece instanceof King)) {
      return false;
    }

    const whiteCastlingFrom: Square = { column: 'E', row: 1 };
    const whiteLongCastlingTo: Square = { column: 'C', row: 1 };
    const whiteShortCastlingTo: Square = { column: 'G', row: 1 };
    const blackCastlingFrom: Square = { column: 'E', row: 8 };
    const blackLongCastlingTo: Square = { column: 'C', row: 8 };
    const blackShortCastlingTo: Square = { column: 'G', row: 8 };

    const castlingFrom = chosenPiece.side === Side.WHITE ? whiteCastlingFrom : blackCastlingFrom;
    const longCastlingTo = chosenPiece.side === Side.WHITE ? whiteLongCastlingTo : blackLongCastlingTo;
    const shortCastlingTo = chosenPiece.side === Side.WHITE ? whiteShortCastlingTo : blackShortCastlingTo;
    return this.isSameSquare(squareFrom, castlingFrom) && (this.isSameSquare(squareTo, longCastlingTo) || this.isSameSquare(squareTo, shortCastlingTo));
  }
```

Reply:
Albo jeszcze dalej :) 
```.ts
  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
    const chosenPiece = this.board.onPositionPiece(squareFrom);
    if (!(chosenPiece instanceof King)) {
      return false;
    }
    
    const castling = {
      WHITE: {
        from: { column: 'E', row: 1 } as Square,
        longTo: { column: 'C', row: 1 } as Square,
        shortTo: { column: 'G', row: 1 } as Square
      },
      BLACK: {
        from: { column: 'E', row: 8 } as Square,
        longTo: { column: 'C', row: 8 } as Square,
        shortTo: { column: 'G', row: 8 } as Square
      }
    }

    return this.isSameSquare(squareFrom, castling[chosenPiece.side].from) 
      && (this.isSameSquare(squareTo, castling[chosenPiece.side].longTo) || this.isSameSquare(squareTo, castling[chosenPiece.side].shortTo));
  }
```

Reply:
dona

Reply:
A może jeszcze lepiej - przez switcha? Wtedy kolor sprawdzamy tylko w jednym miejscu.

Reply:
Switch ma ten problem, ze nie zwraca wartości i wtedy musisz wprowadzić najpierw zmienna niezainicjalizowana i to jeszcze let a nie cost. Wątpię, czy to polepszy czytelność :) 
Możesz sprawdzić raz ten kolor robić tak np: 
`const sideCastling = castling[chosenPiece.side]` i potem odnoszac sie do tego.

Reply:
Dobra, to niech już zostanie :P